substrate/client/src/error.rs

```diff
@@ -111,6 +118,12 @@ error_chain! {
 			description("remote fetch failed"),
 			display("Remote data fetch has been failed"),
 		}
+
+		/// Error decoding extrinsic outcome.
+		ApplyExtinsicFailed {
```

Comment:
1. I'm not sure why this is called `ApplyExtinsicFailed`? I think it is more like `ApplyResultDecodeFailed`
2. Why the comment, description and display messages refers to an `outcome` and not to a `result`?